kvui.py

```diff
@@ -326,48 +327,67 @@ def refresh_view_attrs(self, rv, index, data):
         self.finding_text = data["finding"]["text"]
         self.location_text = data["location"]["text"]
         self.entrance_text = data["entrance"]["text"]
-        self.found_text = data["found"]["text"]
+        self.status_text = data["status"]["text"]
+        self.hint = data["status"]["hint"]
         self.height = self.minimum_height
         return super(HintLabel, self).refresh_view_attrs(rv, index, data)
 
     def on_touch_down(self, touch):
         """ Add selection on touch down """
         if super(HintLabel, self).on_touch_down(touch):
             return True
-        if self.index:  # skip header
-            if self.collide_point(*touch.pos):
-                if self.selected:
-                    self.parent.clear_selection()
+        alt: bool = touch.button == "right" or touch.is_double_tap
+        if self.collide_point(*touch.pos):
+            if self.index:  # skip header
+                if alt:
+                    if self.hint["status"] == HintStatus.HINT_FOUND:
+                        return
+                    ctx = App.get_running_app().ctx
+                    if ctx.slot == self.hint["receiving_player"]:  # If this player owns this hint
+                        status_chain: typing.Dict[HintStatus, HintStatus] = {
+                            HintStatus.HINT_UNSPECIFIED: HintStatus.HINT_NO_PRIORITY,
+                            HintStatus.HINT_NO_PRIORITY: HintStatus.HINT_AVOID,
+                            HintStatus.HINT_AVOID: HintStatus.HINT_PRIORITY,
+                            HintStatus.HINT_PRIORITY: HintStatus.HINT_NO_PRIORITY,
+                        }
+                        ctx.update_hint(self.hint["location"],
+                                        self.hint["finding_player"],
+                                        status_chain.get(self.hint["status"], HintStatus.HINT_UNSPECIFIED))
```

Comment:
Hmmm. I don't know how possible this is, but I really think there should be some sort of "delay" here, where if the user rapidly cycles through the statuses. it's only actually sent once they stop for a couple of seconds. This seems like a lot of messages to the server, and high ping would make this annoying to use

Reply:
A dropdown menu on rclick similar to how I handled it in GodotAP would probably work even better 
![image](https://github.com/user-attachments/assets/1ea139f0-592a-4bb3-b782-fb655dc5f489)
however I lack the kivy wizardry to attempt such

Reply:
> So maybe that can be motivating to try and figure it out, idk

motivation doesn't help tendinitis in both wrists, sadly. So like I said if someone else wants to do that...